w08h03/test/pgdp/datastructures/UnitTest.java

```diff
@@ -36,6 +42,90 @@ <Type extends Comparable<Type>> void testArray(Type[] input) {
 				"Invalid Iteration Count. Expected [" + expected.length + "] got [" + position + "]");
 	}
 
+	@Test
+	@DisplayName("Read 1M nodes")
+	public void testPerformance1M() {
+		int amountOfValues = (int) 1e6;
+
+		List<Integer> values = IntStream.range(0, amountOfValues).boxed().collect(Collectors.toList());
+		List<Integer> solution = IntStream.range(0, amountOfValues).boxed().collect(Collectors.toList());
+		Collections.shuffle(values);
```

Comment:
Der Iterator sollte eigentlich nur die Elemente nacheinander auslesen, so wie sie in der überliegenden Liste enthalten sind. Sortieren ist nicht seine Aufgabe, was dein shuffle aber impliziert.

Oder verstehe ich etwas falsch und du wolltest etwas anderes damit bezwecken?

Reply:
Hast recht, shufflen ist hier egal. Am Ende entsteht der gleiche Baum, aber ich wollte klar machen dass das in beliebiger Reihenfolge geschiehen kann.

Reply:
Also wenn wir `shuffle` benutzen, dann am beste auch wieder mit einem Seed, damit der Test wirklich immer identisch ist.

Man kann bei `shuffle` auch sein eigenes `Random` Objekt übergeben.

```suggestion
		Collections.shuffle(values, new Random(seed));
```

Oder auch einfach `Collections.shuffle(values)` entfernen, da ich wie @TheAleWu finde, dass dies nicht die Aufgabe von diesem Test ist. Die einzige Aufgabe von diesem Test ist ein Benchmark zu sein.

Reply:
Not shuffling causes a stack overflow in the insert function. I will add a seed.